micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -780,6 +815,7 @@ public Config commonTags(String... tags) {
         public synchronized Config meterFilter(MeterFilter filter) {
             if (!meterMap.isEmpty()) {
                 logWarningAboutLateFilter();
+                stalePreFilterIds.addAll(preFilterIdToMeterMap.keySet());
```

Comment:
Should we synchronize this on `meterMapLock`? Given that meterfilter updation is very very rare, this could potentially avoid any meter creation-related activities when filters are getting updated WDYT.